src/main.cpp

```diff
@@ -140,14 +126,17 @@ static bool _get_options(int argc, char **argv, Options &options)
 		("name,n", po::value<string>(&options.name),
 		 "The name of the device as it should appear in the dashboard. If not specified, it will use the device's UUID")
 
+		("factory,f", po::value<string>(&options.factory)->default_value(osrelease.factory),
```

Comment:
I think we shouldn't allow customers to register a device to any factory, at least in customer friendly way.
I understand that token permissions can prevent from device registration at the factories that a given device/user is not part of. But, they still can register the same device to factories they have access to.
Maybe, we should not allow factory overriding if `osrelease.factory` is present by the command line parameter. Only, allow it by the current mean - through the env variable `DEVICE_FACTORY`.

Also, the tuf root meta pre-provisioning won't let user to run aklite against any factory anyway. So, I am not sure if it makes sense in introduction of the `--factory` parameter at all.

Using `osrealease.factory` makes sense to me. If it's not present then I would check for `DEVICE_FACTORY` env variable, if the one is missing then just exit with error code.

Reply:
This is part of a larger effort we are tracking internally - https://foundriesio.atlassian.net/browse/FFTK-2064